edtLHD/src/main/java/fr/univtln/lhd/model/entities/dao/slots/SlotDAO.java

```diff
@@ -0,0 +1,130 @@
+import fr.univtln.lhd.model.entities.dao.DAO;
+import fr.univtln.lhd.model.entities.dao.Datasource;
+import fr.univtln.lhd.model.entities.slots.Classroom;
+import fr.univtln.lhd.model.entities.slots.Group;
+import fr.univtln.lhd.model.entities.slots.Slot;
+import fr.univtln.lhd.model.entities.slots.Subject;
+import fr.univtln.lhd.model.entities.users.Professor;
+import lombok.extern.slf4j.Slf4j;
+import org.threeten.extra.Interval;
+
+import java.sql.*;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+/*
+ */
+public class SlotDAO implements DAO<Slot> {
+    private SlotDAO () {
+    }
+
+    private static final String SELECT_STMT = """
+            select id,date_part('epoch',lower(timerange)) as begin_epoch,
+            date_part('epoch',upper(timerange)) as end_epoch,classroom,memo,subject,type
+            from slots WHERE id = (?)
+            """;
+    //private final String save = "INSERT INTO slots (DEFAULT,(?),(?),(?),(?),(?)";
+    //private final String  update;
+
+    private  static final String DELETE_STMT = "DELETE FROM slots WHERE id=(?) CASCADE";
+
+    public static SlotDAO getInstance () {
+        return new SlotDAO();
+    }
+
+
+    /**
+     * Getter for one Slot
+     *
+     * @param id numerical long identifier for getting the Slot
+     * @return May return one Slot instance
+     */
+    @Override
+    public Optional<Slot> get ( final long id ) throws SQLException {
+        try (Connection conn = Datasource.getConnection();
+             PreparedStatement stmt = conn.prepareStatement(SELECT_STMT)
+        ) {
+            stmt.setLong(1, id);
+            stmt.executeQuery();
+            ResultSet rs = stmt.getResultSet();
+            if (!rs.next()) {
+                return Optional.empty();
+            }
+            long fetchedID = rs.getLong(1);
+            final Instant fetchedBegin = Instant.ofEpochSecond(rs.getLong(2));
+            final Instant fetchedEnd = Instant.ofEpochSecond(rs.getLong(3));
+            final Interval fetchedInterval = Interval.of(fetchedBegin, fetchedEnd);
+            final Classroom fetchedClassroom = ClassroomDAO.getInstance().get(rs.getLong(4)).orElseThrow(SQLException::new);
+            final String fetchedMemo = rs.getString(5);
+            final Subject fetchedSubject = SubjectDAO.getInstance().get(rs.getLong(6)).orElseThrow(SQLException::new);
+            final Slot.SlotType fetchedType = Slot.SlotType.valueOf(rs.getString(7));
+            final List<Group> fetchedGroup = new ArrayList<>(GroupDAO.getInstance().getGroupOfSlot(id)); //the returned list is unmodifiable, requiring a copy.
+            return Optional.of(Slot.getInstance(fetchedID, fetchedType, fetchedClassroom, fetchedSubject, fetchedGroup, List.of(Professor.of("test","test","test@lhd.org","test")),fetchedInterval, fetchedMemo));
+        } catch (SQLException e) {
```

Comment:
if you want to throw an exception, you should goes to one more level of abstraction.
in other word a user using the dao should not see an ``SQLException``

Reply:
Yeah, I wondered whether we should create a Special Exception ("databaseException") or just the SQL error. 
I've left this as is for now.

Reply:
k